--- old/cosyvoice.py	2026-03-13 13:05:05.175583649 +0800
+++ new/cosyvoice.py	2026-03-28 21:54:33.130866698 +0800
@@ -71,6 +71,7 @@
         model_input = self.frontend.frontend_zero_shot('', prompt_text, prompt_wav, self.sample_rate, '')
         del model_input['text']
         del model_input['text_len']
+        model_input['embedding'] = model_input['llm_embedding']
         self.frontend.spk2info[zero_shot_spk_id] = model_input
         return True
 
